contracts/Example2.sol

```diff
@@ -1,11 +1,19 @@
+// SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
 contract Example2 {
     uint public counter;
  
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+    function incrementBy(uint[] calldata arr) public {
```

Comment:
This can be marked as external

Reply:
sure but I believe it doesn't impact gas rt?

Reply:
Ideally it should, but I am not sure if compiler somehow optimises this for you